src/elf/dynamic.rs

```diff
@@ -454,8 +455,8 @@ macro_rules! elf_dyn_std_impl {
             impl From<Dyn> for ElfDyn {
                 fn from(dynamic: Dyn) -> Self {
                     ElfDyn {
-                        d_tag: dynamic.d_tag as u64,
-                        d_val: dynamic.d_val as u64,
+                        d_tag: u64::from(dynamic.d_tag),
```

Comment:
`u64::from` is a cast too though, and the important information is that it is a cast that can't overflow. So yes it does look uglier, but the distinction between `from` and `as` is useful. Generally in parsers such as this, `as` casts are often bugs, so it's good to reduce the number of places where `as` is used so that there's less places that might be bugs.